### PR TITLE
Removed Unused Code from the Slice Compiler

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
I started adding better parameter names in the compiler, but while doing so, there was so much unnecessary code I found, I've split it's removal into it's own PR (this one).

In addition to cleaning up stuff, it also fixes the parameter names in `Scanner.l` and `Scanner.cpp`.

### CODE REVIEW

File: cpp/src/Slice/Grammar.y [hunk 1]
```diff
@@ -1453,7 +1450,7 @@ interface_list
 | ICE_VALUE
 {
     currentUnit->error("illegal inheritance from type Value");
-    $$ = make_shared<ClassListTok>(); // Dummy
```
Comment (InsertCreativityHere) [inline cpp/src/Slice/Grammar.y:1456]: This is the only place where we use `ClassListTok` and it makes no sense.
The name of this grammar rule is `interface_list`, and the caller immediately casts this to an `InterfaceListTok` anyways.

File: cpp/src/Slice/GrammarUtil.h [hunk 2]
```diff
@@ -10,189 +10,79 @@
...
-    class BoolTok;
-    class IntegerTok;
-    class FloatingTok;
-    class ExceptionListTok;
-    class ClassListTok;
-    class InterfaceListTok;
-    class EnumeratorListTok;
-    class ConstDefTok;
-    class OptionalDefTok;
-    class ClassIdTok;
```
Comment (InsertCreativityHere) [inline cpp/src/Slice/GrammarUtil.h:26]: Is there any black-magic-C++ reason to forward declare all these super-simple classes?

File: cpp/src/Slice/Parser.h [hunk 3]
```diff
@@ -104,7 +104,6 @@ namespace Slice
     bool containedEqual(const ContainedPtr& lhs, const ContainedPtr& rhs);
 
     using TypeList = std::list<TypePtr>;
-    using ExceptionList = std::list<ExceptionPtr>;
```
Comment (InsertCreativityHere) [inline cpp/src/Slice/Parser.h:107]: If you expand the few lines below this, you'll see that we double-declare `ExceptionList`.
I guess we wanted to make really sure it got set.

**Comment by InsertCreativityHere** [comment]
I've now made a possibly controversial change in this PR:
I told `clang-format` to ignore the generated Bison compiler code.

1) The generated Bison code is not meant for humans to read. It is an insane tangle of jump tables and archaic variables.
Keeping this code well-formatted is of no value, and has a negligible effect on it's total readability.
2) It's a mild pain to format this code, since 2 separate files must be kept in sync, and `clang-format` only touches one of them.
The process is:
    - run `clang-format` and see what it changed in `Grammar.cpp`
    - Manually replicate these changes in the `Grammar.y` file
    - Regenerate the `Grammar.cpp` file since `Grammar.y` was changed.
    - re-run `clang-format` over the newly regenerated `Grammar.cpp` file
    - Sometimes you have to repeat this process an additional time

**Comment by InsertCreativityHere** [comment]
What we really want is a way to format the `Grammar.y` file, so the generated `Grammar.cpp` would be correctly formatted for free.
But I doubt there's a nice way to format the C++ code weaved throughout the `Grammar.y` file though...

**Review by externl — APPROVED** [review]

**Pull request merged**